### PR TITLE
[AMORO-3189] Support loading mixed catalog without ams url on Spark

### LINKED ISSUE #3189: [Subtask]: Support loading mixed catalog without AMS dependency on Spark
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### Description

Support loading mixed catalog without the AMS URL on Spark

### Parent issue

#3188 

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhoujinsong**
## Why are the changes needed?

Close #3189.

## Brief change log

- Support loading MixedFormatSparkCatalog without the AMS URL
- Support loading MixedFormatSparkSessionCatalog without the AMS URL
- Support loading SparkUnifiedCatalog without the AMS URL
- Support loading SparkUnifiedSessionCatalog without the AMS URL

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Review by Aireed — APPROVED** [review]
hi， i left some tiny comment. PTAL

File: docs/engines/spark/spark-conf.md [hunk 1]
```diff
@@ -31,13 +31,22 @@ menu:
...
 spark.sql.catalog.mixed_catalog=org.apache.amoro.spark.MixedFormatSparkCatalog
 spark.sql.catalog.mixed_catalog.url=thrift://${AMS_HOST}:${AMS_PORT}/${AMS_CATALOG_NAME_HIVE}
 ```
 
+Or create a mixed_catalog with local configurations with following properties:
+```properties
+spark.sql.catalog.mixed_catalog=org.apache.amoro.spark.MixedFormatSparkCatalog
+# Configure mixed catalog type as you needed
+spark.sql.catalog.mixed_catalog.type=hadoop
+spark.sql.catalog.default.warehouse=/warehouse/hadoop_mixed_catalog
```
[the remark this answers is not in the document]
Comment (zhoujinsong) [reply docs/engines/spark/spark-conf.md:47]: Yes, I will fix it.

[the remark this answers is not in the document]
Comment (zhoujinsong) [reply docs/engines/spark/spark-conf.md]: Fixed.

**Comment by zhoujinsong** [comment]
Thanks for the review! @Aireed

**Pull request merged**